include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -276,8 +292,51 @@ public :
                             return analyse_on_offsetted;
                         }
                     }
-#else
-                    analyse_result code = check_segment(previous, current, turn, false);
+                }
+            }
+        }
+
+        // It is nowhere outside, and not on segment, so it is within
+        return analyse_within;
+    }
+
+};
+
+template <>
+class analyse_turn_wrt_point_piece<false>
+{
+public :
+    template <typename Turn, typename Piece>
+    static inline analyse_result apply(Turn const& turn, Piece const& piece)
+    {
+        typedef typename Piece::section_type section_type;
+        typedef typename Turn::robust_point_type point_type;
+        typedef typename geometry::coordinate_type<point_type>::type coordinate_type;
+
+        typedef strategy::within::winding<point_type> strategy_type;
```

Comment:
Actually now there are separate winding strategies for each coordinate system. This one is there for backward compatibility and internally calls one of the CS specific ones based on PointType, see: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/agnostic/point_in_poly_winding.hpp#L63. So my guess is that you should get this strategy from the one passed into the buffer.

Reply:
Done, can you verify if this is what you intended?